stl/inc/vector

```diff
@@ -458,6 +496,135 @@ public:
     using difference_type = typename _Alty_traits::difference_type;
 
 private:
+#ifdef _INSERT_VECTOR_ANNOTATION
+    class _NODISCARD _Asan_annotator {
+    public:
+        _Asan_annotator(const _Asan_annotator&) = delete;
+        _Asan_annotator& operator=(const _Asan_annotator&) = delete;
+
+        explicit constexpr _Asan_annotator(vector* _Myvec_, size_type _Count_) noexcept
+            : _Myvec(_Myvec_), _Count(_Count_) {
+            _STL_INTERNAL_CHECK(_Myvec != nullptr);
+            auto& _My_data = _Myvec->_Mypair._Myval2;
+            _Apply_annotation(_My_data._Myfirst, _My_data._Myend, _My_data._Mylast, _My_data._Mylast + _Count);
+        }
+
+        _CONSTEXPR20 ~_Asan_annotator() {
+            if (!_Myvec) { // Operation succeeded, no modification to the shadow memory required.
+                return;
+            }
+
+            // Shrinks the shadow memory to the current size of the vector.
+            auto& _My_data = _Myvec->_Mypair._Myval2;
+            _Apply_annotation(_My_data._Myfirst, _My_data._Myend, _My_data._Mylast + _Count, _My_data._Mylast);
+        }
+
+        _CONSTEXPR20 void _Release() noexcept {
+            _Myvec = nullptr;
+        }
+
+        static _CONSTEXPR20 void _Create_annotation(const vector& _Myvec_) noexcept {
+            // Annotates the shadow memory of the valid range
+            auto& _My_data = _Myvec_._Mypair._Myval2;
+            _Apply_annotation(_My_data._Myfirst, _My_data._Myend, _My_data._Myend, _My_data._Mylast);
+        }
+
+        static _CONSTEXPR20 void _Clear_annotation(const vector& _Myvec_) noexcept {
+            // Removes the shadow memory annotation of the range
+            auto& _My_data = _Myvec_._Mypair._Myval2;
+            _Apply_annotation(_My_data._Myfirst, _My_data._Myend, _My_data._Mylast, _My_data._Myend);
+        }
+
+        static _CONSTEXPR20 void _Extend_annotation(const vector& _Myvec_, const size_type _Count) noexcept {
+            // Extends the annotated range by _Count
+            auto& _My_data = _Myvec_._Mypair._Myval2;
+            _Apply_annotation(_My_data._Myfirst, _My_data._Myend, _My_data._Mylast, _My_data._Mylast + _Count);
+        }
+
+        static _CONSTEXPR20 void _Shrink_annotation(const vector& _Myvec_, const size_type _Count) noexcept {
+            // Shrinks the annotated range by _Count
+            auto& _My_data = _Myvec_._Mypair._Myval2;
+            _Apply_annotation(_My_data._Myfirst, _My_data._Myend, _My_data._Mylast, _My_data._Mylast - _Count);
+        }

```

Comment:
It was initially. however, we get into trouble if the there are Operations that change more that `numeric_limits<difference_type>::max`

Reply:
Per [tab:container.req], for a container `a`, `a.size()` is `distance(a.begin(), a.end())`, and `a.max_size()` is `distance(x.begin(), x.end())` where `x` is "the largest possible container." Since `distance(x.begin(), x.end())` would have undefined behavior if `x.end()` was more than `numeric_limits<difference_type>::max()` increments from `x.begin(), `numeric_limits<difference_type>::max()` is an upper bound on the size of the largest possible container.

Reply:
Didn't realize that when I suggested to Miscco that we change it.